src/spec-configuration/containerCollectionsOCI.ts

```diff
@@ -3,19 +3,19 @@ import * as semver from 'semver';
 import * as tar from 'tar';
 import * as jsonc from 'jsonc-parser';
 
-import { request } from '../spec-utils/httpRequest';
 import { Log, LogLevel } from '../spec-utils/log';
 import { isLocalFile, mkdirpLocal, readLocalFile, writeLocalFile } from '../spec-utils/pfs';
+import { requestEnsureAuthenticated } from './httpOCIRegistry';
 
 export const DEVCONTAINER_MANIFEST_MEDIATYPE = 'application/vnd.devcontainers';
 export const DEVCONTAINER_TAR_LAYER_MEDIATYPE = 'application/vnd.devcontainers.layer.v1+tar';
 export const DEVCONTAINER_COLLECTION_LAYER_MEDIATYPE = 'application/vnd.devcontainers.collection.layer.v1+json';
 
-export type HEADERS = { 'authorization'?: string; 'user-agent': string; 'content-type'?: string; 'accept'?: string; 'content-length'?: string };
 
 export interface CommonParams {
 	env: NodeJS.ProcessEnv;
 	output: Log;
+	cachedAuthHeader?: string;
```

Comment:
Should this be a map from registries to auth headers?

Reply:
Good point. For the push operations I think this would be fine (we push to one registry at a time), but for pulling perhaps this `params` object is reused? Nonetheless makes sense to make it a map to be sure we never send an auth header to the wrong server.